File: content-solutions/greedy/rope-lift.java
import java.util.*;

class Solution {
    public int solution(int[] ropes) {
        int[] r = ropes.clone();
        Arrays.sort(r);
        int n = r.length, best = 0;
        for (int k = 1; k <= n; k++) best = Math.max(best, r[n - k] * k);
        return best;
    }
}

File: content-solutions/greedy/rope-lift.js
function solution(ropes) {
  const r = [...ropes].sort((a, b) => b - a);
  let best = 0;
  for (let k = 1; k <= r.length; k++) best = Math.max(best, r[k - 1] * k);
  return best;
}
